src/main/java/org/jabref/logic/cleanup/PageFieldCleanup.java

```diff
@@ -0,0 +1,34 @@
+package org.jabref.logic.cleanup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.model.FieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+/**
+ * adds the article ID of a journal as the page count, but only if the page field is empty
+ */
+public class PageFieldCleanup implements CleanupJob {
```

Comment:
Since this is only used for the DOI fetcher, I would suggest to add this functionality as a private class method in `DoiFetcher` instead of a new class.